tests/unittests/tests-cbor/tests-cbor.c

```diff
@@ -279,6 +279,39 @@ static void test_byte_string(void)
     }
 }
 
+static void test_byte_string_no_copy(void)
+{
+    char buffer[128];
+
+    {
+        const char *input = "";
+        unsigned char data[] = {0x40};
+        unsigned char *out;
+        size_t size;
+        TEST_ASSERT(cbor_serialize_byte_string(&stream, input));
+        CBOR_CHECK_SERIALIZED(stream, data, sizeof(data));
+        TEST_ASSERT(cbor_deserialize_byte_string_no_copy(&stream, 0, &out, &size));
+        memcpy(buffer, out, size);
+        buffer[size] = '\0';
+        CBOR_CHECK_DESERIALIZED(input, buffer, EQUAL_STRING);
+    }
```

Comment:
Embracing the context allows us to reuse the same names but I think it uses stack memory until the function is finished.
Maybe we could clean this up a bit and lower stack usage in a followup PR.